pkg/registry/controller/strategy.go

```diff
@@ -55,12 +55,10 @@ func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
-	// TODO: once RC has a status sub-resource we can enable this.
-	//newController := obj.(*api.ReplicationController)
-	//oldController := old.(*api.ReplicationController)
-	//newController.Status = oldController.Status
 	newController := obj.(*api.ReplicationController)
 	oldController := old.(*api.ReplicationController)
+	// update is not allowed to set status
+	newController.Status = oldController.Status
```

Comment:
I had raised that concern prior to submitting the PR.

See @bgrant0607 comment here:
https://github.com/kubernetes/kubernetes/issues/4909#issuecomment-143383218

I agree with his analysis.